From 34ced7dd73ebaeb8f72a416235f1b61cd9774e0e Mon Sep 17 00:00:00 2001
From: Fancy Fang <chen.fang@nxp.com>
Date: Tue, 8 Aug 2017 14:34:36 +0800
Subject: [PATCH 2350/5242] MLK-16158-5 video: fbdev: adv7535: refine low
 refresh settings

commit  61802236c8712b339f02b1c6a71ba9ff13ca3b9e from
https://source.codeaurora.org/external/imx/linux-imx.git

Use the 'refresh' in video mode to determine to set
low refresh config for adv7535 wheter or not.

Signed-off-by: Fancy Fang <chen.fang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/video/fbdev/mxc/adv7535.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/video/fbdev/mxc/adv7535.c b/drivers/video/fbdev/mxc/adv7535.c
index c232810..f47378f 100644
--- a/drivers/video/fbdev/mxc/adv7535.c
+++ b/drivers/video/fbdev/mxc/adv7535.c
@@ -216,7 +216,8 @@ static int adv7535_vmode_cfg(struct adv7535_info *info)
 
 #ifdef CONFIG_FB_IMX64
 	/* low refresh rate */
-	adv7535_write_reg(0x4A, 0x8C);
+	if (fb_vmode->refresh < 50)
+		adv7535_write_reg(0x4A, 0x8C);
 #else
 	adv7535_write_reg(0x17, 0x60); /* VS & HS Low Polarity, DE disabled */
 #endif
-- 
1.7.9.5

